prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/CustomFHIRFunctions.kt

```diff
@@ -248,15 +253,37 @@ object CustomFHIRFunctions : FhirPathFunctions {
                 else -> throw SchemaException("Call with day, month, or year")
             }
         } else {
-            calculateAgeWithAssumption(date)
+            calculateAgeWithAssumption(date, comparisonDate)
+        }
+    }
+
+    private fun getParamValue(params: List<Base>): Pair<LocalDate, String?> {
```

Comment:
Definitely. Both the unit of time and the date to compare are optional params. It just seemed easiest to allow them in either position since they may only pass one or both of them and I will already have the logic for checking the first param for both. I will update the method names to make it include the actual function name as well as an explanation. I will also add kdocs for them all

Reply:
Gotcha, yea I think this might be a good 17th, or eng sync topic. This `parameters` setup has always been confusing and I can't quite tell whether it's something we're in control of or having to conform to. But having a list of lists has never made sense to me, and it might be nice to be able to have named parameters or something. Kind of a larger discussion.. As for this PR:

I still think that if we want to support these params in either position based on the value type, this is still a pretty hard-to-read way of doing that, just following the various pairs and having somewhat generic variable names gets confusing. What about doing something like this (pseudocode), either in a helper function or in the main function?

```
if params.size > 2 fail
initialize timePeriodIndicator, comparisonDate
params.forEach{
  when(it.type) {
    DateType -> set comparisonDate, fail if already set
    StringType -> set timePeriodIndicator, fail if already set or invalid value
    else -> fail
  }
}
```

Reply:
Changes made, ready for you again. Thank you for the cleanup, it is really awesome. I think I was too far into the weeds on this one.